src/proto/loaded_image.rs

```diff
@@ -65,6 +65,18 @@ impl LoadedImage {
         }
     }
 
+    /// Set the image data address and size.
+    ///
+    /// # Safety
+    ///
+    /// This function takes `data` as a raw pointer because the data is not
+    /// owned by `LoadedImage`. The caller must ensure that the memory lives
+    /// long enough.
```

Comment:
Could you include the example use case (from shim) in the method docs? It's a very specific scenario, so it's good to have the motivation stored for future reference

Reply:
Added to the docstring